presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/environment/SinglenodePostgresql.java

```diff
@@ -71,7 +71,8 @@ private DockerContainer createPostgreSql()
                 .withEnv("POSTGRES_DB", "test")
                 .withEnv("PGPORT", Integer.toString(POSTGRESQL_PORT))
                 .withStartupCheckStrategy(new IsRunningStartupCheckStrategy())
-                .waitingFor(forSelectedPorts(POSTGRESQL_PORT));
+                .withHealthCheck(resourceProvider.getPath("health-check.sh"))
```

Comment:
Do we need such a think in all other environments?
That will be quite a lot of db-specific health checks there.

Reply:
Maybe please add some rationale (cmt msg or code comment) for this one, so we can better understand the need for other checks in the future

Reply:
Health checks have an additional benefit of being monitored after the container is started and marked as healthy (so if the server dies due to sam internal/external error launcher will know about that). Lots of docker containers have health checks defined. Also "forHealthCheck" wait strategy is more reliable from my experience than waiting for port or log message (server can die right after port is marked as available)

Reply:
Generally, I'm in favor of adding health checks to other containers (if not already defined). I'll check which one do have those.

Reply:
I've added a rationale to commit message

Reply:
Thanks. If we're planning on adding more of them, let's extend `DockerContainer#withHealthCheck` so that we can provide a command, rather than a file. This will simplify declaration of these checks